src/bindings/python/flux/job/list.py

```diff
@@ -97,6 +97,44 @@ def job_list_id(flux_handle, jobid, attrs=[]):
     return rpc
 
 
+# get_job is the single variant of job_list_id, and returns the
+# expected data structure (dict) to describe one job (or None)
+def get_job(flux_handle, jobid):
+    """Get job information dictionary based on a jobid
+
+    This is a courtesy function for the user that looks for it
+    in the flux.job namespace after submitting a job and wanting
+    information about it.
+    """
+    payload = {"id": int(jobid), "attrs": ["all"]}
+    rpc = JobListIdRPC(flux_handle, "job-list.list-id", payload)
+    try:
+        jobinfo = rpc.get()
+
+    # The job does not exist!
+    except FileNotFoundError:
+        return None
```

Comment:
Is there any chance that `None` will be returned if this function is called immediately after submission, or is that guaranteed not to happen? Maybe a question for @grondo

Reply:
as long as the ID is valid it shouldn't happen.  If the `job-list` module has not yet gotten job data to return to the user, it will wait until it is available.